### PR TITLE
[Enhancement] Support Update Strategy for FE/BE/CN component

### PR DESCRIPTION
**Pull request by yandongxiao**
# Description

see https://github.com/StarRocks/starrocks-kubernetes-operator/issues/592

# Checklist

For operator, please complete the following checklist:

- [x] run `make generate` to generate the code.
- [x] run `golangci-lint run` to check the code style.
- [x] run `make test` to run UT.
- [x] run `make manifests` to update the yaml files of CRD.

For helm chart, please complete the following checklist:

- [x] make sure you have updated the [values.yaml](../helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml)
  file of starrocks chart.
- [x] In `scripts` directory, run `bash create-parent-chart-values.sh` to update the values.yaml file of the parent
  chart( kube-starrocks chart).

### CODE REVIEW

**Review by kevincai — APPROVED** [review]

**Comment by dengliu** [comment]
Hi @yandongxiao 
For StatefulSets in version 1.3.0, `.spec.updateStrategy.rollingUpdate.maxUnavailable` is NOT applicable/available.
The maxUnavailable field for StatefulSets was introduced much later - it became a beta feature in Kubernetes 1.24 (released in 2022).
In v1.3.0, StatefulSets only supported:

.spec.updateStrategy.type with values "OnDelete" or "RollingUpdate"
.spec.updateStrategy.rollingUpdate.partition

If you need to control how many pods can be unavailable during updates in v1.3.0, you'll need to use the partition field instead to control the rollout.

File: helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml [hunk 1]
```diff
@@ -683,6 +690,12 @@ starrocksCnSpec:
     #   volumeMounts:
     #   - mountPath: /opt/starrocks/cn/storage
     #     name: cn-data   # append -data to the end of the name of the starrocksCnSpec.storageSpec.name
+  # Max unavailable pods for the fe component when doing rolling update.
```
Comment (dengliu) [inline helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml:693]: s / `fe component`/`cn component`

File: helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml [hunk 2]
```diff
@@ -946,6 +959,12 @@ starrocksBeSpec:
     #   volumeMounts:
     #   - mountPath: /opt/starrocks/be/storage
     #     name: be-data   # append -data to the end of the name of the starrocksBeSpec.storageSpec.name
+  # Max unavailable pods for the fe component when doing rolling update.
```
Comment (dengliu) [inline helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml:962]: s / `fe component`/`be component`

File: helm-charts/charts/kube-starrocks/values.yaml [hunk 3]
```diff
@@ -791,6 +798,12 @@ starrocks:
       #   volumeMounts:
       #   - mountPath: /opt/starrocks/cn/storage
       #     name: cn-data   # append -data to the end of the name of the starrocksCnSpec.storageSpec.name
+    # Max unavailable pods for the fe component when doing rolling update.
```
Comment (dengliu) [inline helm-charts/charts/kube-starrocks/values.yaml:801]: s / `fe component`/`cn component`

File: helm-charts/charts/kube-starrocks/values.yaml [hunk 4]
```diff
@@ -1054,6 +1067,12 @@ starrocks:
       #   volumeMounts:
       #   - mountPath: /opt/starrocks/be/storage
       #     name: be-data   # append -data to the end of the name of the starrocksBeSpec.storageSpec.name
+    # Max unavailable pods for the fe component when doing rolling update.
```
Comment (dengliu) [inline helm-charts/charts/kube-starrocks/values.yaml:1070]: s / `fe component`/`be component`

**Pull request merged**